src/lib/uri/uri.c

```diff
@@ -186,8 +224,8 @@ uri_format(char *str, int len, const struct uri *uri, bool write_password)
 	if (uri->path != NULL) {
 		SNPRINT(total, snprintf, str, len, "%s", uri->path);
 	}
-	if (uri->query != NULL) {
-		SNPRINT(total, snprintf, str, len, "?%s", uri->query);
+	if (uri->params != NULL) {
+		total += uri_format_query_params(&str, &len, uri);
```

Comment:
I don't agree with you here, because without "params" table we can't test this part! Without "params" table, params after uri format is always equal to query!

Reply:
After discussion, we decided to leave it as it is